ksqldb-rest-app/src/main/java/io/confluent/ksql/api/util/ApiServerUtils.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.util;
+
+import io.confluent.ksql.properties.PropertiesUtil;
+import io.confluent.ksql.rest.entity.KsqlRequest;
+import io.confluent.ksql.rest.server.KsqlRestConfig;
+import io.confluent.ksql.util.FileWatcher;
+import io.confluent.ksql.util.FileWatcher.Callback;
+import io.confluent.ksql.util.QueryMask;
+import io.confluent.ksql.util.VertxSslOptionsFactory;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PfxOptions;
+import io.vertx.ext.web.RoutingContext;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.channels.ClosedChannelException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.config.SslConfigs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ApiServerUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(ApiServerUtils.class);
+
+  private ApiServerUtils() {
+  }
+
+  public static void setMaskedSqlIfNeeded(final KsqlRequest request) {
+    try {
+      request.getMaskedKsql();
+    } catch (final Exception e) {
+      ApiServerUtils.setMaskedSql(request);
+    }
+  }
+
+  public static void setMaskedSql(final KsqlRequest request) {
+    request.setMaskedKsql(QueryMask.getMaskedStatement(request.getUnmaskedKsql()));
+  }
+
+  public static void unhandledExceptionHandler(final Throwable t) {
```

Comment:
Same comment as in the 7.2.x backport about only backporting the necessary methods.